Documentation Update

cmsirbu opened the pull request:
## What's changed

- Moved existing documentation into a new structure to prepare for the mkdocs rendered docsite.
- Edited and improved the visuals in various places using new syntax from the [Material for Mkdocs theme](https://squidfunk.github.io/mkdocs-material/reference/).
- Added dependencies to poetry environment for development.
    + `mkdocs-material` is downgraded to `8.3.0` to maintain compatibility for testing with Nautobot 1.2
- Added configuration for building and hosting the docsite on RTD.
- Replaced the project README with a standardized one. Note: the README will have certain links/images broken until the PR is merged into `develop`.
- Added `invoke docs` task to aid serving docs locally in the dev environment.
- Added test to check that RTD used `docs/requirements.txt` are in sync with dependencies in `pyproject.toml`.

## Building docs

You can build the docs locally in the poetry environment:

```
poetry install
poetry run mkdocs serve
```

See below notes on development environment (more specifically the docker based one).

## TODO

Some sections throughout the files listed below require input (or removal if not applicable to this plugin). All `Developer Note` comments should be resolved and removed.
- [ ] migrate all previously written documentation which now resides in the `old.docs` folder
- [ ] review and test admin guide instructions
- [ ] dev/dev_environment.md - boilerplate guide based on cookie template (old GETTING_STARTED guide). A sync to the cookie dev environment would be beneficial to ensure instructions are accurate.
- [ ] dev/code_reference - check examples provided and add any module/package auto-generated docs that are relevant

qduk commented:
I worked with @cmsirbu to pick up where he left off. I'm pretty sure I've got things at least headed in the right direction but would love some reviews. A couple questions that came to mind while finishing this up. 

1. I wasn't sure on the min version and compatibility matrix. Please review.
2. We don't currently have any FAQs on the old docs. Should I remove that page or leave it for future use?
3. Pertaining to the last bullet point in the TODO, I did add the `models.py`. That is all that seemed relevant to me.
4. Is `test admin guide` from the TODO just ensuring the installation/uninstall procedure is accurate?
5. This does, to my knowledge, now have the updated dev environment document. 

@cmsirbu Can you possibly take a quick look before others review and let me know if I'm missing anything?

itdependsnetworks commented:
faqs:

* Is the application actually a Single Source of Truth?

In reality the application intends to have behaviors as if it was a SSoT. The difference being, the application intends to aggregate data in the real world where it is not feasible to have the System of Record be in a single system.

itdependsnetworks commented:
I think it would be good to include jobs as well.

snaselj commented:
Compatibility matrix updated with respect to future version release

cmsirbu commented:
Is the next release cut from `develop` going to be 1.3.0? CC @chadell 

I'm asking because there are conflicts in this branch now for poetry, but also if the minimum Nautobot version moves to 1.4 we don't need to downgrade `mkdocs-material` anymore.

chadell commented:
yes, 1.3.0 will be the next one

cmsirbu commented on docs/user/app_use_cases.md:
```diff
@@ -0,0 +1,89 @@
+
+![Sync logs view](../images/sync_logs.png)
+
+## Use-cases and common workflows
```
Empty section - considering the details in the previous section, if there's no other workflow to showcase, you could remove it.

qduk replied:
Removed!

cmsirbu commented on docs/admin/install.md:
```diff
@@ -0,0 +1,67 @@
+
+## Prerequisites
+
+- The plugin is compatible with Nautobot 1.0.3 and higher.
```
Since `develop` is going to be 1.3.0 I think this needs to be updated as per the compat matrix.

qduk replied:
updated to specify Nautobot 1.4.0.

cmsirbu commented on docs/dev/contributing.md:
```diff
@@ -0,0 +1,46 @@
+
+The branching policy includes the following tenets:
+
+* The develop branch is the primary branch to develop off of
+* PRs intended to add new features should be sourced from the develop branch.
+* PRs intended to address bug fixes and security patches should be sourced from the develop branch.
```
```suggestion
* The `develop` branch is the primary branch to develop off of.
* PRs intended to add new features should be sourced from the `develop` branch.
* PRs intended to address bug fixes and security patches should be sourced from the `develop` branch.
```

qduk replied:
Fixed.

cmsirbu commented on docs/dev/arch_decision.md:
```diff
@@ -0,0 +1,3 @@
+# Architecture Decision Records
```
Remove if N/A (from nav as well).

qduk replied:
Done.

cmsirbu reviewed (commented):
Very nice work @qduk, left a few comments!

cmsirbu commented:
🚨 🚨 🚨 I've rebased the branch to `develop` to fix conflicts and also include all the changes to the environment for CI. 🚨 🚨 🚨

A couple other changes:
- Updated docs dependencies to latest versions.
- Add content edit/view buttons from material-mkdocs 9.0.
- Regenerated poetry.lock

It looks like the Nautobot docker container used for `development/Dockerfile` is using a poetry version that doesn't like the 1.3.x lockfile. I guess this is an upstream fix.

cmsirbu commented:
After some investigation, the poetry/CI failing issue is solved by #70. @chadell let me know if you want me to rebase this on top of #70 or wait until it's merged into develop.

qduk commented:
> I think it would be good to include jobs as well.

I'm not sure I follow. A question in the FAQ about jobs?

cmsirbu commented:
CI is now passing and the branch is on par with develop. This PR is pretty much finished.

chadell reviewed (approved):
LGTM
